src/test/unit_test/FileDirGuard.h

```diff
@@ -0,0 +1,170 @@
+//------------------------------------------------------------------------------
+/*
+This file is part of rippled: https://github.com/ripple/rippled
+Copyright (c) 2012-2015 Ripple Labs Inc.
+
+Permission to use, copy, modify, and/or distribute this software for any
+purpose  with  or without fee is hereby granted, provided that the above
+copyright notice and this permission notice appear in all copies.
+
+THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#ifndef TEST_UNIT_TEST_DIRGUARD_H
+#define TEST_UNIT_TEST_DIRGUARD_H
+
+#include <ripple/basics/contract.h>
+#include <test/jtx/TestSuite.h>
+#include <boost/filesystem.hpp>
+
+namespace ripple {
+namespace test {
+namespace detail {
+
+/**
+    Create a directory and remove it when it's done
+*/
+class DirGuard
+{
+protected:
+    using path = boost::filesystem::path;
+
+private:
+    path subDir_;
+    bool rmSubDir_{false};
+
+protected:
+    beast::unit_test::suite& test_;
+
+    auto rmDir (path const& toRm)
+    {
+        if (is_directory (toRm) && is_empty (toRm))
+            remove (toRm);
+        else
+            test_.log << "Expected " << toRm.string ()
+            << " to be an empty existing directory." << std::endl;
+    }
+
+public:
+    DirGuard (beast::unit_test::suite& test, path subDir,
+        bool useCounter = true)
+        : subDir_ (std::move (subDir))
+        , test_ (test)
+    {
+        using namespace boost::filesystem;
+        {
+            static auto subDirCounter = 0;
+            if (useCounter)
+                subDir_ += std::to_string(++subDirCounter);
+            if (!exists (subDir_))
+            {
+                create_directory (subDir_);
```

Comment:
Consider using `create_directories` instead.

Reply:
I'm not sure that's a good idea. The intention of `DirGuard` is to clean up after itself. All the operatoins are in the "current" directory (so we don't have to worry about it not existing), and only makes folders one level deep at a time. If a deeper path is provided, (eg. `foo/bar/baz`), then `create_directories` will create all three folders, but the `DirGuard` destructor will only delete `baz`, which is not the desired behavior of this class. I'd rather the behavior of `create_directory`, which (I think) will fail.

Reply:
Yeah, providing a deeper path will fail. Good point, I agree keeping it to one level is a good idea.